options/base_options.py

```diff
@@ -1,6 +1,7 @@
-from dataclasses import fields, MISSING, Field, _recursive_repr, _FIELDS, _FIELD, asdict
+from dataclasses import fields, MISSING, Field, _FIELDS, _FIELD, asdict
 from typing import Any, Type, Optional
 import typing
+from reprlib import recursive_repr
```

Comment:
What's the reasoning behind adding another import when recursive_repr already exists in dataclasses?

Reply:
https://github.com/python/cpython/blob/558b517bb571999ef36123635e0245d083a28305/Lib/dataclasses.py#L248 Since dataclasses is just copying the implementation. Importing it from original location makes it easier to identify where to find documentation on it, since you have to go all the way to the Python Source to know where to find the actual documentation.

Reply:
But I don't want documentation on the function, I just want it to behave the same way as it does in dataclasses. The dataclasses implementation and the reprlib implementation seem slightly different, and the point is just to extend dataclasses fields with a subclass, so I copied everything as closely as possible.

Reply:
It also seems `@recursive_repr`'s implementation didn't work with `Field` as well.